polygon/heimdall/service_persistence.go

```diff
@@ -0,0 +1,67 @@
+package heimdall
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon-lib/log/v3"
+	"github.com/ledgerwatch/erigon/polygon/polygoncommon"
+)
+
+type ServicePersistence interface {
+	Checkpoints() EntityStore[*Checkpoint]
+	Milestones() EntityStore[*Milestone]
+	Spans() EntityStore[*Span]
+	Prepare(ctx context.Context) error
+	Close()
+}
+
+func NewMdbxPersistence(logger log.Logger, dataDir string, tmpDir string) *MdbxServicePersistence {
```

Comment:
Not a fan of this name, maybe something like `NewMdbxStore` or `NewMdbxPersistentStore`? I think we may have some nomenclature clash here.

Reply:
renamed to `NewMdbxServiceStore`, `ServiceStore` interface (`Store` interface is already taken by the previous Heimdall implementation - once we remove that I can rename ServiceStore to Store)